tests/mpi/fe_tools_extrapolate_common.h

```diff
@@ -0,0 +1,363 @@
+                  ExcInternalError());
+
+    }
+  if (fe2.degree > fe1.degree)
+    {
+      AssertThrow(global_error_after < global_error_before,
+                  ExcInternalError());
+    }
+  if (Utilities::MPI::this_mpi_process(MPI_COMM_WORLD)==0)
+    deallog << "OK" << std::endl;
+}
+
+
+template <int dim, typename VectorType>
+void
+check_this_dealii (const FiniteElement<dim> &fe1,
+                   const FiniteElement<dim> &fe2)
+{
+  deallog << std::setprecision (10);
+
+  // only check if both elements have support points.
+  // otherwise, interpolation doesn't really work
+  if ((fe1.get_unit_support_points().size() == 0) ||
+      (fe2.get_unit_support_points().size() == 0))
+    return;
+  //  likewise for non-primitive elements
+  if (!fe1.is_primitive() || !fe2.is_primitive())
+    return;
+  // we need to have dof_constraints for this test
+  if (!fe2.constraints_are_implemented())
+    return;
+  // we need prolongation matrices in fe2
+  if (!fe2.isotropic_restriction_is_implemented())
+    return;
+
+  std_cxx11::unique_ptr<parallel::distributed::Triangulation<dim> > tria(make_tria<dim>());
+
+  std_cxx11::unique_ptr<DoFHandler<dim> >    dof1(make_dof_handler (*tria, fe1));
+  std_cxx11::unique_ptr<DoFHandler<dim> >    dof2(make_dof_handler (*tria, fe2));
+  ConstraintMatrix cm1;
+  DoFTools::make_hanging_node_constraints (*dof1, cm1);
+  cm1.close ();
+  ConstraintMatrix cm2;
+  DoFTools::make_hanging_node_constraints (*dof2, cm2);
+  cm2.close ();
+
+  IndexSet locally_owned_dofs1 = dof1->locally_owned_dofs();
+  IndexSet locally_relevant_dofs1;
+  DoFTools::extract_locally_relevant_dofs (*dof1, locally_relevant_dofs1);
+  IndexSet locally_owned_dofs2 = dof2->locally_owned_dofs();
+  IndexSet locally_relevant_dofs2;
+  DoFTools::extract_locally_relevant_dofs (*dof2, locally_relevant_dofs2);
+
+  VectorType in_ghosted (locally_owned_dofs1, locally_relevant_dofs1, MPI_COMM_WORLD);
+  VectorType out_ghosted (locally_owned_dofs2, locally_relevant_dofs2, MPI_COMM_WORLD);
+  VectorType out_reference (locally_owned_dofs2, locally_relevant_dofs2, MPI_COMM_WORLD);
+
+  // Choose some reference function of sufficiently high polynomial degree.
+  TestFunction<dim> function (std::max(fe1.degree,fe2.degree));
+  VectorTools::interpolate (*dof1, function, in_ghosted);
+  cm1.distribute(in_ghosted);
+  in_ghosted.update_ghost_values();
+  VectorTools::interpolate (*dof2, function, out_reference);
+  cm2.distribute(out_reference);
+  out_reference.update_ghost_values();
+
+  Vector<double> difference_before;
+  VectorTools::integrate_difference (*dof1, in_ghosted, function, difference_before,
+                                     QGauss<dim>(fe1.degree+2), VectorTools::L2_norm);
+  const double local_error_before = difference_before.l2_norm();
+  const double global_error_before
+    = std::sqrt(Utilities::MPI::sum(std::pow(local_error_before,2.), MPI_COMM_WORLD));
+
+  /*output_vector<dim, VectorType> (in_ghosted,
```

Comment:
I don't feel strongly about it, but got corrected by others to remove commented out code in the past. A middle-ground might be what we do with an ``#ifdef`` in tests/hp/hp_constraints_neither_dominate_01.cc